### PR TITLE
Add E2E Sanity Test for EMF 

### PR DESCRIPTION
**Pull request by khanhntd**
# Description of the issue
Add E2E Sanity Test for EMF 

# Description of changes
* Added integration test to validate that the agent publishes metrics for the EMF input plugin

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Run integration  test on my fork https://github.com/khanhntd/amazon-cloudwatch-agent/actions/runs/3619143298/jobs/6101327470
```
null_resource.integration_test (remote-exec): 2022/12/05 12:29:17 ==============================
null_resource.integration_test (remote-exec): 2022/12/05 12:29:17 ==============EMF==============
null_resource.integration_test (remote-exec): 2022/12/05 12:29:17 ==============Successful==============
null_resource.integration_test (remote-exec): EMFCounter   Successful
```

### CODE REVIEW

**Comment by SaxyPandaBear** [comment]
This includes too many other changes that aren't related to EMF.

File: test/metric/net.go [hunk 1]
```diff
@@ -20,10 +20,11 @@ type NetMetricValueFetcher struct {
...
 
 func (f *NetMetricValueFetcher) Fetch(namespace, metricName string, stat Statistics) (MetricValues, error) {
-	dims := f.getMetricSpecificDimensions()
-	values, err := f.fetch(namespace, metricName, dims, stat)
+	dimensions := f.getMetricSpecificDimensions()
+	dimensions = append(dimensions, f.getInstanceIdDimension())
+	values, err := f.fetch(namespace, metricName, dimensions, stat)
 	if err != nil {
-		log.Printf("Error while fetching metric value for %s: %v", metricName, err)
+		log.Printf("Error while fetching metric value for %v: %v", metricName, err.Error())
```
Comment (sky333999) [inline test/metric/net.go:27]: %s: %s ?

**Comment by sky333999** [comment]
LGTM

File: test/metric/net.go (hunk 1 above)
Comment (khanhntd) [reply test/metric/net.go:27]: Have not fixed this ... yet (only fixing statsd and collectd). But nice catch

**Review by SaxyPandaBear — APPROVED** [review]

**Review by taohungyang — APPROVED** [review]

**Pull request merged**